state-chain/pallets/cf-swapping/src/tests/fees.rs

```diff
@@ -120,80 +111,220 @@ fn test_buy_back_flip() {
-			InternalSwapMinimumNetworkFee::<Test>::set(minimum_network_fee);
-		} else {
-			NetworkFee::set(Permill::from_percent(network_fee_percent));
-			MinimumNetworkFee::<Test>::set(minimum_network_fee);
-			InternalSwapNetworkFee::<Test>::set(Permill::from_percent(0));
-			InternalSwapMinimumNetworkFee::<Test>::set(0);
-		}
+fn swap_uses_correct_network_fee() {
+	const NORMAL_SWAP_AMOUNT: AssetAmount = 10000;
+	const NORMAL_NETWORK_FEE: Permill = Permill::from_percent(10);
+	const NORMAL_MINIMUM_NETWORK_FEE: AssetAmount = 100;
 
-		let collected_fees_before = CollectedNetworkFee::<Test>::get();
+	const INTERNAL_SWAP_AMOUNT: AssetAmount = 20000;
+	const INTERNAL_NETWORK_FEE: Permill = Permill::from_percent(20);
+	const INTERNAL_MINIMUM_NETWORK_FEE: AssetAmount = 200;
 
-		// Create a swap request with the given params
-		SwapRequests::<Test>::insert(
-			SWAP_REQUEST_ID,
-			SwapRequest {
-				id: SWAP_REQUEST_ID,
-				input_asset: Asset::Usdc,
-				output_asset: Asset::Eth,
-				state: SwapRequestState::UserSwap {
-					refund_params: None,
-					output_action: if is_internal {
-						SwapOutputAction::CreditOnChain { account_id: ALICE }
-					} else {
-						SwapOutputAction::Egress {
+	new_test_ext()
+		.execute_with(|| {
+			// Set both network fees to different values
+			NetworkFee::set(NORMAL_NETWORK_FEE);
+			MinimumNetworkFee::<Test>::set(NORMAL_MINIMUM_NETWORK_FEE);
+			InternalSwapNetworkFee::<Test>::set(INTERNAL_NETWORK_FEE);
+			InternalSwapMinimumNetworkFee::<Test>::set(INTERNAL_MINIMUM_NETWORK_FEE);
+
+			// Set a swap rate of 1 to make it easier
+			SwapRate::set(1_f64);
+
+			// Sanity check collected fees before any swaps
+			assert_eq!(CollectedNetworkFee::<Test>::get(), 0);
+
+			// Normal swap
+			fn normal_swap(amount: AssetAmount) {
+				Swapping::init_swap_request(
+					Asset::Flip,
+					amount,
+					Asset::Eth,
+					SwapRequestType::Regular {
+						output_action: SwapOutputAction::Egress {
 							ccm_deposit_metadata: None,
-							output_address: ForeignChainAddress::Eth(Default::default()),
-						}
+							output_address: ForeignChainAddress::Eth(H160::zero()),
+						},
 					},
-					dca_state: DcaState {
-						status: DcaStatus::ChunkScheduled(SwapId(1)),
-						// Not relevant for this test
-						remaining_input_amount: 0,
-						remaining_chunks: 0,
-						chunk_interval: 2,
-						accumulated_output_amount: 0,
-						// Use the given params
-						network_fee_collected,
-						accumulated_stable_amount,
+					Default::default(),
+					None,
+					None,
+					SwapOrigin::Vault {
+						tx_id: TransactionInIdForAnyChain::Evm(H256::default()),
+						broker_id: Some(BROKER),
 					},
-					broker_fees: Default::default(),
-				},
-			},
-		);
+				);
+			}
+			// Normal swap with network fee
+			normal_swap(NORMAL_SWAP_AMOUNT);
+			// Normal swap that will be charged the minimum network fee
+			normal_swap(NORMAL_MINIMUM_NETWORK_FEE);
```

Comment:
It is a bit confusing that you use the fee itself as the amount. Might be useful to show that even swapping, say 5x the `NORMAL_SWAP_AMOUNT` would still result in NORMAL_SWAP_AMOUNT charged in fees.